Mathlib/Order/Concept.lean

```diff
@@ -282,6 +307,36 @@ theorem codisjoint_extent_intent [IsTrichotomous α r'] [IsTrans α r'] :
   · contradiction
   · assumption
 
+theorem isCompl_extent_intent [IsStrictTotalOrder α r'] (c' : Concept α α r') :
+    IsCompl c'.extent c'.intent :=
+  ⟨c'.disjoint_extent_intent, c'.codisjoint_extent_intent⟩
+
+/-- See `isLowerSet_extent'` for the theorem on a `<`-concept. -/
+theorem isLowerSet_extent {α : Type*} [Preorder α] (c : Concept α α (· ≤ ·)) :
+    IsLowerSet c.extent :=
+  @mem_extent_of_rel_extent _ _ _ _
+
+/-- See `isUpperSet_intent'` for the theorem on a `<`-concept. -/
+theorem isUpperSet_intent {α : Type*} [Preorder α] (c : Concept α α (· ≤ ·)) :
+    IsUpperSet c.intent :=
+  @mem_intent_of_intent_rel _ _ _ _
+
+/-- See `isLowerSet_extent` for the theorem on a `≤`-concept. -/
+theorem isLowerSet_extent' {α : Type*} [PartialOrder α] (c : Concept α α (· < ·)) :
+    IsLowerSet c.extent := by
+  intro a b hb ha
+  obtain rfl | hb := hb.eq_or_lt
```

Comment:
I suspect this can be golfed into a one-liner of some kind.

Reply:
Also, this < stuff: I do wonder if there's a general theorem about the complementary relation or the flipped relation or both here. Probably best not to worry for now.

Reply:
`<` is neither the complement or the swap of `≤`. You do have `x < y ↔ ¬ y ≤ x` in the linear order case, but this theorem works even in the partial order case.

I don't really see how to golf this down; in particular, I don't see how to avoid the casing on whether `b = a` or `b < a`.